portfolio/src/main/webapp/script.js

```diff
@@ -29,73 +29,138 @@ function addRandomGreeting() {
+ * @return {function} a function which has @param {string} punctuation
+ *                    which then greets a person with {name} upon being called 
+ */
 function greet(name) {
   return (punctuation) => `Hello ${name}${punctuation}`;
 }
 
+/**
+ * Returns a happier version of an inputted string
+ * @param {string} item - a body of text
+ * @return {string} @param item with the addition of a laughing emoji
+ */
 function addHappiness(item) {
   return item + ' 😂';
 }
 
+/**
+ * Updates the max number of comments which are showing
+ * @return {void}
+ */
 async function updateMaxComments() {
   const maxComments = document.querySelector("#maxComments--number-input").value;
-  defaultMaxComments = maxComments; 
+  defaultMaxComments = maxComments;
   await getData(maxComments);
 }
 
+/**
+ * Deletes all comments which are stored in the datastore
+ */
 async function deleteComments() {
-    try {
-        const ret = await fetch('/delete-data', {
-            method: "POST"
-        })
-        alert("A truly sad day, your comments were deleted");
-        await getData();
-    } catch (e) {
-        alert("Something went wrong in deleting your comments");
-    }
+  try {
+    const ret = await fetch('/delete-data', {
+      method: "POST"
+    })
+    alert("A truly sad day, your comments were deleted");
+    await getData();
+  } catch (e) {
+    alert("Something went wrong in deleting your comments");
+  }
+}
+
+/**
+ * Get the comments from the server and populate the comment DIV
+ * @param {number?} maxComments - the max number of comments to display
+ */
+async function getData(maxComments) {
+  if (!maxComments) maxComments = defaultMaxComments;
+  try {
+    const ret = await fetch('/data?maxComments=' + maxComments);
+    const comments = await ret.json();
+    const commentDiv = document.querySelector('#comments');
+    commentDiv.innerHTML = "";
+    comments.forEach((comment, i) =>
+      commentDiv.innerHTML += `<p>Comment ${i + 1} is ${comment}</p>`);
+  } catch (e) {
+    alert(`Hey!! There was an error: ${e?.message || e}`);
+  }
 }
 
-/** a map of the countries with the highest happiness score */
+/** Class representing a map of the countries with the highest happiness score */
 class HappinessChart {
+  /**
+   * Create the happiness chart
+   * @param {string} divID - The ID of the div which will contain the chart 
+   */
   constructor(divID) {
     google.charts.load('current', {
-      'packages':['geochart'],
+      'packages': ['geochart'],
       'mapsApiKey': 'AIzaSyCQ2pBJaaQ0JIitodS6-sbHfie2Qosefxg'
     });
     google.charts.setOnLoadCallback(async () => {
       const happiestCountries = await this.getHappiestCountries();
-      console.log(happiestCountries)
       const data = this.buildDataTable(happiestCountries);
       this.drawRegionsMap(divID, data);
     });
   }
 
+  /** 
+   * Get the happiest countries from happiest_countries.json
+   * @return {[
+     [string, float]
```

Comment:
optional: might be safer to store these results in an object instead, so that the fields are named in case you ever choose to expand to more fields 
instead of havingto know that arr[0] is name and arr[1] is score, you can parse the results into an object
{name: <country>, score: <score>}